go/src/koding/klientctl/endpoint/remoteapi/remoteapi.go

```diff
@@ -1,39 +1,154 @@
 package remoteapi
 
 import (
+	"errors"
 	"sync"
+	"time"
 
 	"koding/api"
 	"koding/klientctl/config"
+	"koding/klientctl/ctlcli"
 	"koding/klientctl/endpoint"
 	"koding/klientctl/endpoint/kloud"
+	"koding/klientctl/endpoint/team"
 	"koding/remoteapi"
 	"koding/remoteapi/client"
+	account "koding/remoteapi/client/j_account"
+	"koding/remoteapi/models"
 )
 
+// ErrNotFound is returned by Client methods when a requested
+// resource was not found.
+var ErrNotFound = errors.New("resource was not found")
+
+// DefaultClient is a default remote.api client.
+//
+// It uses default kloud client as auth provider
+// and kd cache for storing authentication tokens.
 var DefaultClient = &Client{}
 
+// DefaultTimeout defines max remote.api request time,
+// after which the request is cancelled.
+var DefaultTimeout = 30 * time.Second
+
+// Client is a wrapper for remote.api client that takes
+// care of authorization and caching.
 type Client struct {
-	Kloud *kloud.Client
+	Kloud  *kloud.Client  // if nil, kloud.DefaultClient is used
+	Team   *team.Client   // if nil, team.DefaultClient is used
+	Client *client.Koding // if nil, new client is created (with kloud as auth provider)
 
-	once sync.Once // for c.init()
-	c    *remoteapi.Client
+	once     sync.Once // for c.init()
+	api      *remoteapi.Client
+	c        *client.Koding
+	accounts map[string]*models.JAccount
 }
 
+// New creates new remoteapi client for the given user.
 func (c *Client) New(user *api.User) *client.Koding {
 	c.init()
 
-	return c.c.New(user)
+	return c.api.New(user)
+}
+
+// Close flushes all underlying caches and closes all clients.
+func (c *Client) Close() (err error) {
+	if len(c.accounts) != 0 {
+		err = c.kloud().Cache().SetValue("jAccounts", c.accounts)
+	}
+	return err
+}
+
+// AccountByUsername looks up the jAccount by the given username.
+func (c *Client) AccountByUsername(username string) (*models.JAccount, error) {
+	c.init()
+
+	// TODO(rjeczalik): Make a wrapper type for c.accounts and move the lookup there.
+	for _, account := range c.accounts {
```

Comment:
Entire `AccountByUsername` and `Account` methods use `account` token in a staggered manner. (L73 vs L84)